### DISCUSSION TITLE
v6.0.0 - UDiFF format bhavcopy

### DISCUSSION THREAD

**Discussion by BennyThadikaran** [BennyThadikaran/eod2#152] [Announcements]
EOD2 has been updated to use the new UDiFF bhavcopy format. **From 8th July 2024, old bhavcopy format will be discontinued.**

Make sure you update EOD2 to v6.0.0 and update `nse` package.

- To update EOD2 repo, follow the [update instructions here](https://github.com/BennyThadikaran/eod2/wiki/Installation#updating-the-git-repo)
- To update `nse` package, run `pip install -U nse` (pip3 for linux/mac)

For those curious about UDiFF format, See [NSE circular 62142](https://nsearchives.nseindia.com/content/circulars/MSD62142.zip) dated May 22, 2024 - zip file

TL;DR: 

- Column headers have changed and the file name format is `BhavCopy_NSE_CM_0_0_0_{YYMMDD}_F_0000.csv.zip`
- The `F` in the file name indicated Final or can be Provisional `P`. I suppose the final bhavcopy is what we'd be interested in. 
- If the file exceed 10 million rows, a second file would be included in the zip. This is not a problem for Equity bhavcopy which has about 2800 odd rows.

**Comment by Prady04** [comment]
Hey thanks for this proactive update. really appreciate it

**Comment by cotton1234** [comment]
Thanks Benny, Will update tmrw and check.
